src/covering.ml

```diff
@@ -797,7 +799,7 @@ let env_of_rhs evars ctx env s lets =
     in
       pats, x @ y, List.length x 
   in
-  let pats = List.map (lift (-letslen)) pats @ map (lift len) patslets in
+  let pats = List.map (lift (-letslen)) pats @ patslets in
```

Comment:
@mattam82 It seems that lifting `patslets` was wrong. Since I do not fully understand all this code, I guess it's better that you take a look at it...